staging/src/k8s.io/apiserver/pkg/server/serve.go

```diff
@@ -93,23 +95,31 @@ func (s *GenericAPIServer) serveInsecurely(stopCh <-chan struct{}) error {
 	}
 	glog.Infof("Serving insecurely on %s", s.InsecureServingInfo.BindAddress)
 	var err error
-	s.effectiveInsecurePort, err = runServer(insecureServer, s.InsecureServingInfo.BindNetwork, stopCh)
+	s.effectiveInsecurePort, err = runServer(insecureServer, s.InsecureServingInfo, stopCh)
 	return err
 }
 
 // runServer listens on the given port, then spawns a go-routine continuously serving
 // until the stopCh is closed. The port is returned. This function does not block.
-func runServer(server *http.Server, network string, stopCh <-chan struct{}) (int, error) {
+func runServer(server *http.Server, info *ServingInfo, stopCh <-chan struct{}) (int, error) {
 	if len(server.Addr) == 0 {
 		return 0, errors.New("address cannot be empty")
 	}
 
-	if len(network) == 0 {
-		network = "tcp"
+	if len(info.BindNetwork) == 0 {
+		info.BindNetwork = "tcp"
+	}
+	server.ConnState = func(conn net.Conn, state http.ConnState) {
+		if state == http.StateClosed || state == http.StateHijacked {
```

Comment:
In fact StateHijacked is a terminal state in the http server code, so they specifically must not count against the limit. That's why this code decrements when the connection goes in to the hijack state.